tests/integration/ha_tests/test_smoke.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from asyncio import TimeoutError
+
+import pytest
+from juju import tag
+from pytest_operator.plugin import OpsTest
+from tenacity import Retrying, stop_after_delay, wait_fixed
+
+from ..helpers import (
+    APPLICATION_NAME,
+    CHARM_SERIES,
+    get_primary,
+)
+from .helpers import (
+    check_db,
+    check_password_auth,
+    create_db,
+    get_any_deatached_storage,
+    is_postgresql_ready,
+    is_storage_exists,
+    storage_id,
+)
+
+TEST_GARBADGE_STORAGE_NAME = "test_pgdata"
+TEST_DATABASE_RELATION_NAME = "test_database"
+DUP_APPLICATION_NAME = "postgres-test-dup"
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_removal(ops_test: OpsTest, charm: str):
```

Comment:
IMHO we don't need this test. The default storage pool for LXD, rootfs, will be deleted when released and destroying the volume is not something that the charm handles.